packages/material-ui/src/Drawer/Drawer.js

```diff
@@ -131,29 +192,37 @@ const Drawer = React.forwardRef(function Drawer(props, ref) {
   }, []);
 
   const anchor = getAnchor(theme, anchorProp);
+
+  const styleProps = {
```

Comment:
add open, elevation and other missing properties, because you limiting the use of  variant functionality in createMuiTheme components overrides